### ISSUE TITLE
Autocomplete doesn't work anymore 

### ISSUE THREAD

**Issue by domaschh** [zed-industries/community#1790]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When I try to apply a suggested autocomplete word with tab or enter it just either inserts a new line or tabs. Clicking the suggestion works tho.

### Environment

Zed: v0.95.2 (stable)
OS: macOS 13.2.1
Memory: 64 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

[logs.txt](https://github.com/zed-industries/community/files/12104146/logs.txt)

**Comment by rsheidow** [comment]
I have been experiencing this a bit. It hasn't been a consistent thing, but once it starts, it happens for a bit and I just stop using autocomplete.

I have attached an example of what I see:
https://github.com/zed-industries/community/assets/92095340/495bbcd5-ff44-4c29-8856-567107e78f12

I am unsure what causes it, I have been unable to replicate the toggle between it working, and not working.

Environment

Zed: v0.95.3
OS: macOS 13.4.1
Memory: 32GB
Architecture: not sure? it's an M1 Pro lol


Finding this error in the logs:

2023-07-21T22:51:58 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/gpui/src/app.rs:728: JSON RPC method not found: "ResolveCompletionItem" not yet implemented

[Zed.log](https://github.com/zed-industries/community/files/12135051/Zed.log)

**Comment by JosephTLyons** [comment]
Hey @domaschh and @rsheidow - does this happen with all programming languages you use in Zed or just specific ones?

**Comment by rsheidow** [comment]
I can confirm it happens with Golang. I haven’t been using Zed with anything other than that

**Comment by domaschh** [comment]
> I can confirm it happens with Golang. I haven’t been using Zed with anything other than that

It's happening with Rust as well... Rust analyzer correctly shows the completions and suggests them just using tab and enter doesn't work .. clicking on the works.

**Comment by minarc** [comment]
I'm currently using Zed preview version 0.97.1, and I've noticed that Golang's autocomplete feature is sometimes working fine, but it slows down significantly after just a few seconds.

**Comment by minarc** [comment]
https://github.com/zed-industries/community/assets/11865340/a6fd9474-b69e-4edd-a155-fae9680972df

Here's the video. As you can observe, I'm typing `http.StatusTeapot`, but the red line trail very slowly.

**Comment by gcurtis** [comment]
@minarc's video matches exactly what I've been seeing for a while now (at least since June). Out of the languages I use, it only seems to happen with gopls. I haven't seen it with the JSON/YAML language servers.

A little more context in case it helps:

- I use the same exact version of gopls (0.13.1) with Sublime Text and I don't see any of the performance issues there.
- When all diagnostic errors/warnings are fixed the lag seems to go away. As soon as I introduce a syntax error (even in the middle of typing a word), the lag comes back.
- I've tried a bunch of differrent gopls settings to see if anything helps, but haven't had any luck. My current settings.json has gopls configured with:
```json
"lsp": {
  "gopls": {
    "gofumpt": true,
    "initialization_options": {
      "directoryFilters": [
        "-**/node_modules",
        "-**/.git"
      ],
      "gofumpt": true,
      "symbolScope": "workspace",
      "staticcheck": true,
      "templateExtensions": [
        "tmpl"
      ],
      "analyses": {
        "nilness": true,
        "unusedparams": true,
        "unusedvariable": true,
        "unusedwrite": true,
        "useany": true
      },
      "codelenses": {
        "gc_details": true
      }
    }
  }
}
```
- These are the LSP logs I see when experiencing the lag:
```
2023/08/04 11:29:48 imports fixes: AllImportsFixes: .../internal/git/git_test.go:9:6: expected ';', found Mirror (and 2 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:48 imports fixes: AllImportsFixes: .../internal/git/git_test.go:9:6: expected ';', found Mirror (and 2 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:50 imports fixes: AllImportsFixes: .../internal/git/git_test.go:11:1: expected operand, found '}' (and 7 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:50 imports fixes: AllImportsFixes: .../internal/git/git_test.go:11:1: expected operand, found '}' (and 7 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:50 imports fixes: AllImportsFixes: .../internal/git/git_test.go:11:1: expected operand, found '}' (and 7 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:50 imports fixes: AllImportsFixes: .../internal/git/git_test.go:11:1: expected operand, found '}' (and 7 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:50 imports fixes: AllImportsFixes: .../internal/git/git_test.go:10:9: expected ';', found '!' (and 6 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:50 imports fixes: AllImportsFixes: .../internal/git/git_test.go:10:9: expected ';', found '!' (and 6 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:50 imports fixes: AllImportsFixes: .../internal/git/git_test.go:11:1: expected operand, found '}' (and 7 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:50 imports fixes: AllImportsFixes: .../internal/git/git_test.go:11:1: expected operand, found '}' (and 7 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:50 imports fixes: AllImportsFixes: .../internal/git/git_test.go:11:1: expected operand, found '}' (and 7 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:50 imports fixes: AllImportsFixes: .../internal/git/git_test.go:11:1: expected operand, found '}' (and 7 more errors)
	file=".../internal/git/git_test.go"
2023/08/04 11:29:54 no highlight: column is beyond end of line
2023/08/04 11:30:21 background imports cache refresh starting
2023/08/04 11:30:22 background refresh finished after 510.739292ms
2023/08/04 11:31:15 background imports cache refresh starting
2023/08/04 11:31:16 background refresh finished after 411.994292ms
```
- Zed: 0.98.1 (preview)
- OS: macOS 13.5 (22G74)
- Memory: 64 GiB
- Arch: aarch64

**Comment by minarc** [comment]
The attached video above indicates that the autocomplete feature appears very slow, but in reality, it's even worse. I've observed that the autocomplete function is having difficulties with prefix matching.

I haven't tested any languages other than Go. However, React seems to perform well.

**Comment by domaschh** [comment]
Does the autocomplete work for you with using tab and enter? Because that's the main issue for me. Speed for suggestions is okay but I simply cannot accept them with keyboard input. I've tried binding other keys and reinstalling it.

**Comment by domaschh** [comment]
> Does the autocomplete work for you with using tab and enter? Because that's the main issue for me. Speed for suggestions is okay but I simply cannot accept them with keyboard input. I've tried binding other keys and reinstalling it.

https://github.com/zed-industries/community/assets/64631939/6ecb325b-8f37-4a50-bfb5-62f6591bfe53

**Comment by minarc** [comment]
@domaschh Nope, since autocomplete is very slow I can not accept them until they complete.
